calico-vpp-agent/connectivity/vxlan.go

```diff
@@ -146,17 +173,24 @@ func (p *VXLanProvider) AddConnectivity(cn *common.NodeConnectivity) error {
 			Vni:            p.getVXLANVNI(),
 			DecapNextIndex: p.ip4NodeIndex,
 		}
+		if cn.Vni != 0 {
+			tunnel.Vni = cn.Vni
+		} else {
+			return nil
```

Comment:
I think this can be cleaned up now - would be good to confirm we can use VXLAN with extra networks now that we have a proper ipam